iiab-install

```diff
@@ -145,26 +145,43 @@ if [ "$STAGE" -lt 2 ] && [ "$1" == "--debug" ]; then
 fi
 
 # TEMPORARY: Catch images up to current code to benefit from pre-installed apps
+# Workaround for (web-published) images; will go away later
+# Assumes /etc/iiab/iiab_state.yml is not created until (prior run of) Stage 4 but
+# config_vars2.yml is present with the stage counter altered by pi-gen to be 2.
 if [ -f /etc/iiab/config_vars2.yml ]; then
     mv /etc/iiab/config_vars2.yml $IIAB_STATE_FILE
-fi
 
-# Assumes /etc/iiab/iiab_state.yml is not created until (prior run of) Stage 4.
-if [ -f $IIAB_STATE_FILE ]; then
-    # Workaround for (web-published) images; will go away later
+# Fix up prior values in state file
+# mongo role improved post image creation
     if grep -q sugar $IIAB_STATE_FILE; && ! grep -q mongodb $IIAB_STATE_FILE; then
         echo "mongodb_installed: True" >> $IIAB_STATE_FILE
     fi
 
+# another change to accout for
+    sed -i -e 's/pan_bluetooth/bluetooth/' $IIAB_STATE_FILE
+
     if [ "$STAGE" -eq 2 ]; then
         echo -e "\nCompleting Stage 3 from IIAB image (starts systemd service iiab-setup-db to run the 'mysql' role)."
         systemctl start iiab-setup-db
     fi
 
     PLAYBOOK="iiab-from-console.yml"    # Stage 4-9 then Network Role
     ARGS=""    # Removes '--extra-vars reinstall=True' if --reinstall, BUT WHY?
+# the same as --reinstall execpt stage 3 is not run as there are no other functional
+# changes in stage 3 to account for post image creation once the above is run.
+# reinstall=True would force kiwix to re-download and re-install in commit
+# ce2ec3b0cad76449caf3299003b5d297a3164181
+## End image catch up
 fi
 
+# allow iiab-install to read IIAB_STATE_FILE to not repeat installs of previous
+# roles that already completed within the stage.
+if [ ! -f  $IIAB_STATE_FILE ]; then
+    if [ ! -d /etc/iiab ]; then
```

Comment:
I will remove lines 180-182 after merging this into PR #2020, as the existence of /etc/iiab is already being verified further above.

(iiab-install's precondition is very strict and very much on-purpose if /etc/iiab does not exist -- i.e. those who stubbornly insist on not reading any instructions get a careful explanation from Lines 16-37.)